playbooks/common/openshift-cluster/validate_hostnames.yml

```diff
@@ -8,16 +8,27 @@
     register: lookupip
     changed_when: false
     failed_when: false
-  - name: Warn user about bad openshift_hostname values
-    pause:
-      prompt:
+
+  - name: Validate openshift_hostname when defined
+    fail:
+      msg: >
         The hostname {{ openshift.common.hostname }} for {{ ansible_nodename }}
         doesn't resolve to an IP address owned by this host. Please set
         openshift_hostname variable to a hostname that when resolved on the host
         in question resolves to an IP address matching an interface on this
-        host. This host will fail liveness checks for pods utilizing hostPorts,
-        press ENTER to continue or CTRL-C to abort.
-      seconds: "{{ 10 if openshift_override_hostname_check | default(false) | bool else omit }}"
+        host. This host will fail liveness checks for pods utilizing hostPorts.
+        See https://docs.openshift.org/latest/install_config/install/advanced_install.html#configuring-host-variables
     when:
     - lookupip.stdout != '127.0.0.1'
     - lookupip.stdout not in ansible_all_ipv4_addresses
+    - not openshift_override_hostname_check | default(false)
+
+  - name: Validate openshift_ip exists on node when defined
+    fail:
+      msg: >
+        openshift_ip should be set to an IP of an interface on this node
+        Inventory setting: openshift_ip={{ openshift_ip }}
+        See https://docs.openshift.org/latest/install_config/install/advanced_install.html#configuring-host-variables
+    when:
+    - openshift_ip is defined
+    - openshift_ip not in ansible_all_ipv4_addresses
```

Comment:
maybe also make it overridable? (I'm not sure there's a good argument for making the hostname check overridable but NOT the IP check.)